tests/extras/datasets/pandas/test_avro_dataset.py

```diff
@@ -0,0 +1,192 @@
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import PurePosixPath
+
+import numpy
+import pandas
+import pytest
+from fsspec.implementations.http import HTTPFileSystem
+from fsspec.implementations.local import LocalFileSystem
+from gcsfs import GCSFileSystem
+from s3fs.core import S3FileSystem
+
+from kedro.extras.datasets.pandas import AVRODataSet
+from kedro.extras.datasets.pandas.avro_dataset import infer_pandas_dtype_to_avro_type
+from kedro.io import DataSetError
+from kedro.io.core import PROTOCOL_DELIMITER
+
+FILENAME = "test.avro"
+
+
+@pytest.fixture
+@pytest.mark.parametrize("tmp_path", ["/tmp"], indirect=True)
+def filepath_avro(tmp_path):
+    return (tmp_path / FILENAME).as_posix()
+
+
+@pytest.fixture
+def dummy_data():
+    return pandas.DataFrame({"col1": [1, 2], "col2": [4, 5], "col3": [5, 6]})
+
+
+@pytest.fixture
+def avro_pd_data_set(filepath_avro):
+    return AVRODataSet(filepath=filepath_avro)
+
+
+class TestPandasAvroDataSet:
+    def test_credentials_propagated(self, mocker):
+        """Test propagating credentials for connecting to GCS"""
+        mock_fs = mocker.patch("fsspec.filesystem")
+        credentials = {"key": "value"}
+
+        AVRODataSet(filepath=FILENAME, credentials=credentials)
+
+        mock_fs.assert_called_once_with("file", **credentials)
+
+    def test_save_and_load(self, tmp_path, dummy_data):
+        """Test saving and reloading the data set."""
+        filepath = (tmp_path / FILENAME).as_posix()
+        data_set = AVRODataSet(filepath=filepath)
+        data_set.save(dummy_data)
+        reloaded = data_set.load()
+        assert dummy_data.equals(reloaded)
+
+        data_set = AVRODataSet(filepath=tmp_path.as_posix())
+        pattern = r"Saving AVRODataSet to a directory is not supported."
+        with pytest.raises(DataSetError, match=pattern):
+            data_set.save(dummy_data)
+
+    def test_exists(self, avro_pd_data_set, dummy_data):
+        """Test `exists` method invocation for both existing and
+        nonexistent data set."""
+        assert not avro_pd_data_set.exists()
+        avro_pd_data_set.save(dummy_data)
+        assert avro_pd_data_set.exists()
+
+        def patch_error():
+            raise DataSetError("test")
+
+        avro_pd_data_set._get_load_path = patch_error
+        assert not avro_pd_data_set.exists()
```

Comment:
Similar comment to the other test_avro_dataset, it's not very clear what you're testing here so could you move it to a separate test with an explanatory name?